DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -414,6 +428,84 @@ public Task ModifyPositionAsync(int position, string reason = null)
+        /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+        public Task ModifyParentAsync(ulong? parent_id = null, bool? lock_permissions = null, string reason = null)
+        {
+            if (this.Guild == null)
+                throw new InvalidOperationException("Cannot modify parent of non-guild channels.");
+            if (this.IsCategory)
+                throw new InvalidOperationException("Cannot modify parent of category channels.");
+
+            var position = this.Guild._channels.Values.Where(xc => xc.Type == this.Type && xc.ParentId == parent_id).Select(xc => xc.Position).DefaultIfEmpty(-1).Max() + 1;
+            var chns = this.Guild._channels.Values.Where(xc => xc.Type == this.Type).OrderBy(xc => xc.Position).ToArray();
+            var pmds = new RestGuildChannelNewParentPayload[chns.Length];
+            for (var i = 0; i < chns.Length; i++)
+            {
+                pmds[i] = new RestGuildChannelNewParentPayload
+                {
+                    ChannelId = chns[i].Id,
+                    Position = chns[i].Position >= position ? chns[i].Position + 1 : chns[i].Position,
+                };
+                if(chns[i].Id == this.Id)
+                {
+                    pmds[i].Position = position;
+                    pmds[i].ParentId = parent_id;
+                    pmds[i].LockPermissions = lock_permissions;
+                }
+            }
+
+            return this.Discord.ApiClient.ModifyGuildChannelParentAsync(this.Guild.Id, pmds, reason);
+        }
+
+        /// <summary>
+        /// Moves the channel out of a category.
+        /// </summary>
+        /// <param name="reason">Reason for audit logs.</param>
+        /// <returns></returns>
+        /// <exception cref="Exceptions.UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageChannels"/> permission.</exception>
+        /// <exception cref="Exceptions.NotFoundException">Thrown when the channel does not exist.</exception>
+        /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
+        /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+        public Task RemoveParentAsync(string reason = null)
+        {
+            if (this.Guild == null)
+                throw new InvalidOperationException("Cannot modify parent of non-guild channels.");
+            if (this.IsCategory)
+                throw new InvalidOperationException("Cannot modify parent of category channels.");
+
+            var position = this.Guild._channels.Values.Where(xc => xc.Type == this.Type).Select(xc => xc.Position).DefaultIfEmpty(-1).Max() + 1;
+            var chns = this.Guild._channels.Values.Where(xc => xc.Type == this.Type).OrderBy(xc => xc.Position).ToArray();
```

Comment:
Some line breaks would be nice here too.

Reply:
Done, let me know if it's more readable @uwx.